### PR TITLE
feat: add edit teller component

### PR DESCRIPTION
**Pull request by spider0061**
add edit teller component.

## Related issues and discussion
#596

## Screenshots, if any
Screenshot from 2020-05-14 12-00-25

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by karantakalkar** [comment]
@spider0061 please resolve merge conflicts and make same changes as in create teller component.

**Comment by karantakalkar** [comment]
@spider0061 resolve mcs this looks easy merge

File: src/app/organization/tellers/edit-teller/edit-teller.component.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      this.officeData = data.offices;
+    });
+    if (this.tellerData.status) {
+      if (this.tellerData.status === 'ACTIVE') {
+          this.tellerData.status = 300;
+      } else {
+          this.tellerData.status = 400;
+      }
+    }
+    this.tellerStatusesData = [ {'id': 300, 'code': '300', 'value': 'Active'}, {'id': 400, 'code': '400', 'value': 'Inactive'}];
```
Comment (karantakalkar) [inline src/app/organization/tellers/edit-teller/edit-teller.component.ts:60]: wrap this

File: src/app/organization/tellers/edit-teller/edit-teller.component.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (karantakalkar) [inline src/app/organization/tellers/edit-teller/edit-teller.component.ts:61]:
```suggestion
```

File: src/app/organization/tellers/edit-teller/edit-teller.component.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+   */
+  constructor(private formBuilder: FormBuilder,
+              private organizationService: OrganizationService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe((data: { teller: any }) => {
+      this.tellerData = data.teller;
+    });
+    this.route.data.subscribe((data: { offices: any }) => {
```
Comment (karantakalkar) [inline src/app/organization/tellers/edit-teller/edit-teller.component.ts:50]: remove this and emrge into previous route data subscription

**Comment by spider0061** [comment]
@karantakalkar made the changes.

**Comment by muskankhedia** [comment]
@abhaychawla @karantakalkar For the non-editable component, should we prefer for the `mat-form-field` in the inactive state so as to maintain the consistency within the application.

File: src/app/organization/tellers/edit-teller/edit-teller.component.html [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            <input matInput required formControlName="name">
+            <mat-error *ngIf="tellerForm.controls.name.hasError('required')">
+              Teller Name is <strong>required</strong>
+            </mat-error>
+            <mat-error *ngIf="tellerForm.controls.name.hasError('pattern')">
+              Teller Name <strong>cannot</strong> begin with a special character or number
+            </mat-error>
+          </mat-form-field>		
+
+          <mat-label>Office</mat-label>
```
Comment (karantakalkar) [inline src/app/organization/tellers/edit-teller/edit-teller.component.html:22]: @spider0061 replace this with disabled form fields then we will merge

**Comment by spider0061** [comment]
@karantakalkar I tried disabling it but I am getting following error.
Screenshot from 2020-05-14 09-17-53

**Comment by muskankhedia** [comment]
> @karantakalkar I tried disabling it but I am getting the following error on submit.
> Screenshot from 2020-05-14 09-17-53

That's because of the `officeId` key missing in the response. You need to manually add another parameter in the `teller` variable before submitting it. It will work then.

**Comment by karantakalkar** [comment]
see add loan charge form in loans module it contains disabled fields implement similarly this one and edit tax PR.

**Comment by spider0061** [comment]
@karantakalkar done

**Pull request merged**